pkg/controller/clusterinstallation/mattermost.go

```diff
@@ -223,75 +256,60 @@ func (r *ReconcileClusterInstallation) updateMattermostDeployment(mi *mattermost
 	// we will return and not upgrade the deployment.
 	if update {
 		reqLogger.Info(fmt.Sprintf("Running Mattermost image %s upgrade job check", imageName))
-
-		updateName := "mattermost-update-check"
-		job := &batchv1.Job{
-			ObjectMeta: metav1.ObjectMeta{
-				Name:      updateName,
-				Namespace: mi.GetNamespace(),
-			},
-			Spec: batchv1.JobSpec{
-				Template: corev1.PodTemplateSpec{
-					ObjectMeta: metav1.ObjectMeta{
-						Labels: map[string]string{"app": updateName},
-					},
-					Spec: *new.Spec.Template.Spec.DeepCopy(),
-				},
-			},
+		alreadyRunning, err := r.fetchRunningUpdateJob(mi, reqLogger)
+		if err != nil && k8sErrors.IsNotFound(err) {
+			reqLogger.Info("Launching update job")
+			if err = r.launchUpdateJob(mi, new, imageName, reqLogger); err != nil {
+				reqLogger.Error(err, "Launching update job failed")
```

Comment:
Instead of logging these errors and then returning them, let's just return them with their full context. i.e. `return errors.Wrap(err, "Launching update job failed")`.

The idea is we add Info logs as needed throughout the code, but the returned errors will be logged so we can clean things up by not also separately logging them.